designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala

```diff
@@ -500,21 +501,32 @@ class AkkaHttpBasedRouteProvider(
       deploymentServiceSupplier: Supplier[DeploymentService],
       createAllDeployedScenarioService: ProcessingType => AllDeployedScenarioService,
       processingTypeDataStateFactory: ProcessingTypeDataStateFactory,
+      additionalUIConfigProvider: AdditionalUIConfigProvider,
       sttpBackend: SttpBackend[Future, Any],
-      additionalUIConfigProvider: AdditionalUIConfigProvider
-  )(implicit executionContext: ExecutionContext): Resource[IO, ProcessingTypeDataReload] = {
-    implicit val sttpBackendImplicit: SttpBackend[Future, Any] = sttpBackend
+      executionContext: ExecutionContext,
+  ): Resource[IO, ProcessingTypeDataReload] = {
     Resource
       .make(
         acquire = IO(
-          new ProcessingTypeDataReload(() =>
+          new ProcessingTypeDataReload({ () =>
+            def getDeploymentManagerDependencies(processingType: ProcessingType) = {
```

Comment:
Maybe it's good idea to extract/move this function outside: `prepareProcessingTypeData`?

Reply:
This function uses 3 of the method arguments.
Maybe I'm missing something but I can't see an option to extract it without some strange things like:
a) Multiple parameters lists - one with these 3 parameters and the second one with the `processingType`
b) Extracting a new internal class which takes these 3 parameters in the constructor and the `processingType` as a parameter of the single method

Do you see some better approach?